tensorboard/uploader/uploader_test.py

```diff
@@ -228,6 +236,138 @@ def scalar_event(tag, value):
+            mock_client,
+            "/logs/foo",
+            rpc_rate_limiter=mock_rate_limiter,
+            blob_rpc_rate_limiter=mock_blob_rate_limiter,
+            allowed_plugins=["scalars", "graphs"],
+        )
+        uploader.create_experiment()
+
+        def graph_event(tag, value):
+            return event_pb2.Event(graph_def=value)
+
+        mock_logdir_loader = mock.create_autospec(logdir_loader.LogdirLoader)
+        mock_logdir_loader.get_run_events.side_effect = [
+            {
+                "run 1": [
+                    graph_event("1.1", bytes(950)),
+                    graph_event("1.2", bytes(950)),
+                ],
+                "run 2": [
+                    graph_event("2.1", bytes(950)),
+                    graph_event("2.2", bytes(950)),
+                ],
+            },
+            {
+                "run 3": [
+                    graph_event("3.1", bytes(950)),
+                    graph_event("3.2", bytes(950)),
+                ],
+                "run 4": [
+                    graph_event("4.1", bytes(950)),
+                    graph_event("4.2", bytes(950)),
+                ],
+                "run 5": [
+                    graph_event("5.1", bytes(950)),
+                    graph_event("5.2", bytes(950)),
+                ],
+            },
+            AbortUploadError,
+        ]
+
+        with mock.patch.object(
+            uploader, "_logdir_loader", mock_logdir_loader
+        ), self.assertRaises(AbortUploadError):
+            uploader.start_uploading()
+        self.assertEqual(1, mock_client.CreateExperiment.call_count)
+        self.assertEqual(10, mock_client.WriteBlob.call_count)
+        self.assertEqual(0, mock_rate_limiter.tick.call_count)
+        self.assertEqual(10, mock_blob_rate_limiter.tick.call_count)
+
+    def test_upload_server_error(self):
+        mock_client = _create_mock_client()
+        mock_rate_limiter = mock.create_autospec(util.RateLimiter)
+        mock_blob_rate_limiter = mock.create_autospec(util.RateLimiter)
+        uploader = _create_uploader(
+            mock_client,
+            "/logs/foo",
+            rpc_rate_limiter=mock_rate_limiter,
+            blob_rpc_rate_limiter=mock_blob_rate_limiter,
+            allowed_plugins=["scalars", "graphs"],
+        )
+        uploader.create_experiment()
+
+        def graph_event(tag, value):
+            return event_pb2.Event(graph_def=value)
+
+        mock_logdir_loader = mock.create_autospec(logdir_loader.LogdirLoader)
+        mock_logdir_loader.get_run_events.side_effect = [
+            {"run 1": [graph_event("1.1", bytes(950))],},
+            {"run 1": [graph_event("1.2", bytes(950))],},
+            AbortUploadError,
+        ]
+
+        mock_client.WriteBlob.side_effect = [
+            [write_service_pb2.WriteBlobResponse()],
+            test_util.grpc_error(grpc.StatusCode.INTERNAL, "nope"),
+        ]
+
+        # This demonstrates that the ALREADY_EXISTS error is handled
```

Comment:
Fix comment? This comment applies to the test case below, but not to
this test case.

Reply:
Oops, copypasta, fixed